thunder/transforms/cudagraph.py

```diff
@@ -59,12 +59,17 @@ class CUDAGraphRunner:
     change.
     """
 
-    def __init__(self):
+    def __init__(self, share_mem_pool: bool = False):
         self.cuda_graph_cache = {}  # cahce_key (.make_cache_key) -> (graph, static_inputs, static_outputs)
         self.python_callables = {}  # fn_name -> (callable. static_input_mask (or None))
         self.trace_symbols = {}  # fn_name -> (bsyms, inputs, outputs)
         self.name_counter = 1
 
+        if share_mem_pool:
+            self.global_mem_pool = torch.cuda.graph_pool_handle()
```

Comment:
Nit: I'd prefer if we could call it `mem_pool`, without global. It is specific to the transform/runner (which is good).